### PR TITLE
fix(NODE-6764): incorrect negative bigint handling

### PR DESCRIPTION
**Pull request by W-A-James**
### Description

Correct regression to bigint deserialziation introduced in #649 

### ⚠️ Fixed potential data corruption bug when `useBigInt64` is enabled

After refactoring to improve deserialization performance in #649, we inadvertently broke the deserialization of negative Longs into `bigint`. This issue has been fixed here.

Thanks to @rkistner for reporting this bug!

### Double check the following

- [X] Ran `npm run check:lint` script
- [X] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [X] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [X] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

**Review by rkistner — APPROVED** [review]
Thanks for the quick fix! The change looks good to me, and should fix the issue I had

File: test/node/bigint.test.ts [hunk 1]
```diff
@@ -105,6 +105,39 @@ describe('BSON BigInt support', function () {
 
       it(description, test);
     }
+
+    describe('edge case tests', function () {
```
Comment (durran) [inline test/node/bigint.test.ts:109]: Just a little nit. Instead of just calling these edge case tests can we better describe what we are testing here?

**Review by durran — CHANGES_REQUESTED** [review]

File: src/utils/number_utils.ts [hunk 2]
```diff
@@ -87,10 +87,12 @@ export const NumberUtils: NumberUtils = {
     const hi = NumberUtils.getUint32LE(source, offset + 4);
 
     /*
-      eslint-disable-next-line no-restricted-globals
+      eslint-disable no-restricted-globals
       -- This is allowed since this helper should not be called unless bigint features are enabled
      */
-    return (BigInt(hi) << BigInt(32)) + BigInt(lo);
+    const intermediate = (BigInt(hi) << BigInt(32)) + BigInt(lo);
+    return BigInt.asIntN(64, intermediate);
```
Comment (nbbeeken) [inline src/utils/number_utils.ts:94]: From slack, let's fix the bit math instead of using asIntN

**Review by nbbeeken — CHANGES_REQUESTED** [review]

**Review by nbbeeken — APPROVED** [review]
